cmd/tailscale/cli/debug.go

```diff
@@ -807,6 +818,49 @@ func runDaemonBusEvents(ctx context.Context, args []string) error {
 	return nil
 }
 
+var daemonBusGraphArgs struct {
+	format string
+}
+
+func runDaemonBusGraph(ctx context.Context, args []string) error {
+	graph, err := localClient.GetBusGraph(ctx)
+	if err != nil {
+		return err
+	}
+	if daemonBusGraphArgs.format == "dot" {
+		var topics []eventbus.DebugTopic
+		json.Unmarshal([]byte(graph), &topics)
+		dotGraph := generateDOTGraph(topics)
+		fmt.Print(dotGraph)
+	} else {
+		fmt.Print(string(graph))
+	}
+	return nil
+}
+
+// generateDOTGraph generates the DOT graph format based on the events
+func generateDOTGraph(topics []eventbus.DebugTopic) string {
+	var sb strings.Builder
+	sb.WriteString("digraph event_bus {\n")
+
+	for _, topic := range topics {
+		// If topic name is empty, set it to "???"
+		if topic.Name == "" {
+			topic.Name = "???"
+		}
+		// If no subscribers, still ensure the topic is drawn
+		if len(topic.Subscribers) < 1 {
+			topic.Subscribers = append(topic.Subscribers, "no-subscribers")
+		}
+		for _, subscriber := range topic.Subscribers {
+			sb.WriteString(fmt.Sprintf("\t\"%s\" -> \"%s\" [label=\"%s\"];\n", topic.Publisher, subscriber, topic.Name))
```

Comment:
Perhaps easier to read?
```suggestion
			fmt.Fprintf(sb, "\t%q -> %q [label=%q];\n", topic.Publisher, subscriber, topic.Name)
```

Reply:
Also if you like: `cmp.Or(topic.Name, "???")` saves a few lines.